scipy/linalg/_special_matrices.py

```diff
@@ -1396,3 +1398,517 @@ def convolution_matrix(a, n, mode='full'):
         col0 = az
         row0 = raz[-n:]
     return toeplitz(col0, row0)
+
+
+class sakurai:
```

Comment:
Why is this a class? All other matrices are constructed by a direct function call. The `__init__` + `callable` combination does the same in principle.

It should IMO not be a class, and if it has to be a class, this would need much more discussion (aside from a capitalized name).

Reply:
I just wanted to have a good looking way to call various objects, like mikota.callable, which the class natively provides. I don't know python well enough to figure out other similar good looking alternatives and of course I am open to constructive specific suggestions.

Reply:
> good looking way

That's extremely subjective. So far, all matrices here can be simply generated with `X = name_of_the_matrix(...)`. I can see canonical matrix pairs as being constructed through `X, Y = name_of_the_pair(...)`, but I don't understand why you want to have something like `mikota.callable`. What's wrong with `mikota(...)`?

> and of course I am open to constructive specific suggestions.

It's you who wants/needs to add these functions; I don't understand your needs as well as you do, and I'm not so familiar with the linalg code overall (but at least I'm trying to give your PRs _some_ review...); that said, my specific suggestion is: barring extremely compelling evidence to the contrary (i.e. not just "good looking"), follow the patterns already existing in that module (resp. how those matrices are used in scipy).

Reply:
@h-vetinari Most if not all iterative solvers in sparse scipy are matrix-free, i.e., do not actually need the matrix stored explicitly in memory, while this is what all the present functions produce in `X = name_of_the_matrix(...)`. E.g., `mikota.callable` is a callable object rather than the Mikota matrix currently produced by `mikota(...)`

Reply:
If you want to introduce a fundamentally new mechanism, a PR is not the place to discuss this. Please open an issue with the proposed design (including what problems it solves, what is necessary to solve them, why the design was chosen over possible alternatives, etc.), then we can discuss if/how we want to support this, and how the API should look. 

It doesn't have to be a specification down the the last detail, but it should at least present the high level requirements and considerations. Your branch can serve as an implementation reference and the benchmark numbers would be supporting evidence that this is worth the maintenance cost.